src/python_inspector/resolution.py

```diff
@@ -150,8 +212,76 @@ def get_requirements_for_package_from_pypi_simple(self, candidate):
                 if dep.scope == "install":
                     yield packaging.requirements.Requirement(str(dep.extracted_requirement))
 
-    def get_requirements_for_package_from_pypi_json_api(self, purl):
+        sdist_file = get_sdist_file(
+            repos=self.repos, candidate=candidate, python_version=python_version
+        )
+
+        if sdist_file:
+            setup_py_path = os.path.join(
+                utils_pypi.CACHE_THIRDPARTY_DIR,
+                "extracted_sdists",
+                sdist_file,
+                sdist_file,
+                "setup.py",
+            )
+            setup_cfg_path = os.path.join(
+                utils_pypi.CACHE_THIRDPARTY_DIR,
+                "extracted_sdists",
+                sdist_file,
+                sdist_file,
+                "setup.cfg",
+            )
+            pkg_info_path = os.path.join(
+                utils_pypi.CACHE_THIRDPARTY_DIR, "extracted_sdists", sdist_file
+            )
+            requirement_path = os.path.join(
+                utils_pypi.CACHE_THIRDPARTY_DIR,
+                "extracted_sdists",
+                sdist_file,
+                sdist_file,
+                "requirements.txt",
+            )
+
+            path_by_sdist_parser = {
+                PythonSdistPkgInfoFile: pkg_info_path,
+                PythonSetupPyHandler: setup_py_path,
+                SetupCfgHandler: setup_cfg_path,
+                PipRequirementsFileHandler: requirement_path,
+            }
+
+            for handler, path in path_by_sdist_parser.items():
+                if not os.path.exists(path):
+                    continue
+
+                deps = list(handler.parse(path))
+                assert len(deps) == 1, handler
+                if not deps:
+                    continue
+                dependencies = deps[0].dependencies
+                for dep in dependencies:
+                    if not dep.purl:
+                        continue
+
+                    dep_purl = PackageURL.from_string(dep.purl)
+                    if not (
+                        dep.scope == "install"
+                        and (
+                            not (dep.is_resolved)
+                            or (dep.is_resolved and dep_purl.name not in self.resolved_requirements)
+                        )
+                    ):
+                        continue
+                    if dep.is_resolved:
+                        self.resolved_requirements = (*self.resolved_requirements, dep_purl)
```

Comment:
```suggestion
                        self.resolved_requirements = self.resolved_requirements + tuple([dep_purl])
```
But IMHO the attribute should be a list not a tuple